javaharness/java/arcs/android/ArcsService.java

```diff
@@ -92,32 +97,39 @@ public void startArc(
         pecPortManager.addPecInnerPortProxy(pecId, pecInnerPortProxy);
 
         runWhenReady(() -> {
-            ArcData.Builder arcDataBuilder = new ArcData.Builder()
+          ArcData.Builder arcDataBuilder = new ArcData.Builder()
               .setArcId(arcId)
               .setPecId(pecId)
               .setRecipe(recipe);
-            for (int i = 0; i < particleIds.size(); ++i) {
-              arcDataBuilder.addParticleData(
-                  new ArcData.ParticleData()
-                      .setId(particleIds.get(i))
-                      .setName(particleNames.get(i))
-                      .setProvidedSlotId(providedSlotIds.get(i)));
+          for (int i = 0; i < particleIds.size(); ++i) {
+            arcDataBuilder.addParticleData(
+                new ArcData.ParticleData()
+                    .setId(particleIds.get(i))
+                    .setName(particleNames.get(i))
+                    .setProvidedSlotId(providedSlotIds.get(i)));
+          }
+
+          ArcData arcData = arcDataBuilder.build();
```

Comment:
I really prefer to keep this in a separate class. 
If the common Arcs interface bothers you, let's remove it for now, fine. But i really like having this in a separate class anyway (if you don't like ArcsLocal, let's make is ArcsHelper or something, but), it is cleaner.

Reply:
Moved into a class called ArcsShellApi.